go/vt/vtgate/buffer/buffer_test.go

```diff
@@ -221,6 +226,24 @@ func waitForState(b *Buffer, want bufferState) error {
 	}
 }
 
+// waitForPoolSlots waits up to 10s that all buffer pool slots have been
+// returned. The wait is necessary because in some cases the buffer code
+// does not block itself on the wait. But in any case, the slot should be
+// returned when the request has finished. See also shardBuffer.unblockAndWait().
+func waitForPoolSlots(b *Buffer, want int) error {
+	start := time.Now()
+	for {
```

Comment:
This is a tight loop and will hang in non-preemptive implementations of the runtime. Even otherwise, it's something we should avoid.